### PR TITLE
Vmware resource pool first add

### PR DESCRIPTION
**Pull request by dav1x**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

new module for vmware_guest.py suite

##### ANSIBLE VERSION

```
-> ansible --version
ansible 2.3.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY

This adds the ability to create resource pools in vmware in vCenter.

```
changed: [localhost] => {
    "changed": true, 
    "invocation": {
        "module_args": {
            "cluster": "devel", 
            "cpu_expandable_reservations": true, 
            "cpu_limit": -1, 
            "cpu_reservation": 0, 
            "cpu_shares": "normal", 
            "datacenter": "Boston", 
            "hostname": "10.x.x.25", 
            "mem_expandable_reservations": true, 
            "mem_limit": -1, 
            "mem_reservation": 0, 
            "mem_shares": "normal", 
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
            "resource_pool": "testrp-final2", 
            "state": "present", 
            "username": "administrator@vsphere.local", 
            "validate_certs": false
        }, 
        "module_name": "vmware_resource_pool"
    }
}


```

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure due to PEP 8 issues:

```
2017-02-08 00:01:28 ERROR: PEP 8: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py:65:161: E501 line too long (190 > 160 characters) (current)
2017-02-08 00:01:28 ERROR: PEP 8: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py:86:161: E501 line too long (190 > 160 characters) (current)
2017-02-08 00:01:28 ERROR: PEP 8: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py:247:11: E111 indentation is not a multiple of four (current)
2017-02-08 00:01:28 ERROR: PEP 8: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py:248:11: E111 indentation is not a multiple of four (current)
2017-02-08 00:01:28 ERROR: PEP 8: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py:251:11: E111 indentation is not a multiple of four (current)
```

**Comment by nerzhul** [comment]
Seems very nice

**Comment by angystardust** [comment]
@dav1x very nice! Any plans to implement a `vmware_datastore.py` module?

**Comment by dav1x** [comment]
@angystardust thanks! What functionality were you thinking for vmware_datastore? I'm planning on doing a vmware_folder next I think.

**Comment by nerzhul** [comment]
it seems there is no idempotency in the module and a fix on vmware_guest from another pr, can you add idempotency ?

**Comment by dav1x** [comment]
I can work on those features next week. My daughter is having surgery today and I'll be at the hospital for the next 5-7 days

**Comment by angystardust** [comment]
@dav1x Thanks for paying attention to my comment. I'm thinking of some basic CRUD operations like "create an nfs or vmfs datastore", "destroy a datastore". You can take some inspiration from the govm tool:
https://github.com/vmware/govmomi/blob/master/govc/USAGE.md#datastorecreate
Do you want me to open a feature-request issue for this? 

P.s.: good luck to you and your daughter for the surgery. Hope it isn't something too serious...

**Comment by dav1x** [comment]
rebased and remove the previous change. @nerzhul I tested the module and it seems to have functional idempotency. You can submit the same play run and it will not repeat nor mark as changed. 

Can you elaborate further as to what is missing?

**Comment by mattclay** [comment]
CI failure due to module docs validation error:

```
2017-02-20 20:17:01 ============================================================================
2017-02-20 20:17:01 lib/ansible/modules/cloud/vmware/vmware_resource_pool.py
2017-02-20 20:17:01 ============================================================================
2017-02-20 20:17:01 TRACE:
2017-02-20 20:17:01     mapping values are not allowed here
2017-02-20 20:17:01       in "vmware_resource_pool.EXAMPLES", line 117, column 11:
2017-02-20 20:17:01             - name: Add resource pool to vCenter
2017-02-20 20:17:01                   ^
2017-02-20 20:17:01 ERROR: EXAMPLES is not valid YAML. Line 117 column 11
```

File: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py
Comment (nerzhul) [inline lib/ansible/modules/cloud/vmware/vmware_resource_pool.py]: why not add spaces around equal ?

File: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py
Comment (nerzhul) [inline lib/ansible/modules/cloud/vmware/vmware_resource_pool.py]: too wide imports, please only import the needed parts or @gundalow will bite you :)

**Review by nerzhul — CHANGES_REQUESTED** [review]
Some changes are required

**Comment by dav1x** [comment]
Hey @nerzhul will add changes today. Thanks for the review!

File: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py
Comment (gundalow) [reply lib/ansible/modules/cloud/vmware/vmware_resource_pool.py]: Hehe,
We will be getting this enforced by CI at somepoint as well, so it's good to get in the habit now :)

File: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py
Comment (mattclay) [inline lib/ansible/modules/cloud/vmware/vmware_resource_pool.py]: CI failure due to PEP 8 issue:

```
2017-02-21 20:33:25 ERROR: PEP 8: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py:322:6: E126 continuation line over-indented for hanging indent (current)
```

You can run the PEP 8 tests locally with `make pep8`.

File: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py
Comment (nerzhul) [inline lib/ansible/modules/cloud/vmware/vmware_resource_pool.py]: can you make the space consisten in your lists ? same for same list above

**Review by nerzhul — CHANGES_REQUESTED** [review]
Except the code style it seems goods

File: lib/ansible/modules/cloud/vmware/vmware_resource_pool.py
Comment (dav1x) [reply lib/ansible/modules/cloud/vmware/vmware_resource_pool.py]: @nerzhul do you mean the spacing by the =?

Comment (nerzhul) [reply lib/ansible/modules/cloud/vmware/vmware_resource_pool.py]: @dav1x yeah

Comment (dav1x) [reply lib/ansible/modules/cloud/vmware/vmware_resource_pool.py]: @nerzhul done!

**Comment by mattclay** [comment]
CI failure due to module validation errors:

```
2017-02-28 17:36:10 ============================================================================
2017-02-28 17:36:10 lib/ansible/modules/cloud/vmware/vmware_resource_pool.py
2017-02-28 17:36:10 ============================================================================
2017-02-28 17:36:10 ERROR:107:Imports should be directly below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 326
2017-02-28 17:36:10 ERROR:107:Imports should be directly below DOCUMENTATION/EXAMPLES/RETURN/ANSIBLE_METADATA. line 328
```

**Comment by dav1x** [comment]
any updates to this? can we take about a merge?

@jctanner @nerzhul

**Comment by dav1x** [comment]
@angystardust I'm going to work on vmware_datastore next. We got the dynamic provisioning in kubernetes 1.5.3 working on vsphere. Programmatically creating datastores would be a great addition to my OpenShift playbooks!

**Comment by jctanner** [comment]
@dav1x were in a feature freeze for 2.3, but this is also waiting on @nerzhul to approve or dismiss his change request.

**Review by nerzhul — COMMENTED** [review]
Code is okay for me, i just need to test it, but as said we are in FF

**Comment by nerzhul** [comment]
I cannot dismiss my review as i'm not marked as an official maintainer for github :(

**Comment by jctanner** [comment]
bot_status

**Comment by dav1x** [comment]
@jctanner @nerzhul any thing we can do for this? Should I remove the PR and re-request it? I've got vmware_folder.py written and ready to contribute now as well.

**Review by nerzhul — APPROVED** [review]
It's okay for me

shipit

**Pull request merged**
